web/src/main/java/com/marklogic/hub/web/service/FlowJobService.java

```diff
@@ -149,6 +153,15 @@ private FlowJobs retrieveJobsByFlowName(String flowName) {
         }
         latestJob.status = jobJson.getString("jobStatus");
 
+        List<String> staleStateJobIds = new ArrayList<>();
+        if (cachedJobsByFlowName.size() == 0) {
+            if (StringUtils.isNotEmpty(latestJob.status) && latestJob.status.startsWith(RUNNING_PREFIX) && StringUtils.isNotEmpty(latestJob.id)) {
+                //invalid state on the job database that may be caused by unexpected server shutdown or crashed
+                staleStateJobIds.add(latestJob.id);
+                latestJob.status = JobStatus.FAILED.toString();
```

Comment:
Are we also considering the possibility that another orchestrator maybe running the job that isn't this client?

Reply:
Yeah. I only considered QS to recover. How can I detect it and work around ?